src/operations.rs

```diff
@@ -684,44 +685,38 @@ fn remove_path_from_specific_file(path: PathBuf) -> Result<()> {
     Ok(())
 }
 
-pub fn find_shell_scripts_to_be_modified(add_case: bool) -> Result<Vec<PathBuf>> {
+pub fn find_shell_scripts_to_be_modified(add_case: bool) -> Vec<PathBuf> {
     let home_dir = dirs::home_dir().unwrap();
 
     let paths_to_test: Vec<PathBuf> = vec![
-        home_dir.join(".bashrc").into(),
-        home_dir.join(".profile").into(),
-        home_dir.join(".bash_profile").into(),
-        home_dir.join(".bash_login").into(),
-        home_dir.join(".zshrc").into(),
+        home_dir.join(".bashrc"),
+        home_dir.join(".profile"),
+        home_dir.join(".bash_profile"),
+        home_dir.join(".bash_login"),
+        home_dir.join(".zshrc"),
     ];
 
-    let result = paths_to_test
+    paths_to_test
         .iter()
         .filter(|p| p.exists() ||
             (add_case && p.file_name().unwrap()==".zshrc" && std::env::consts::OS == "macos") // On MacOS, always edit .zshrc as that is the default shell, but only when we add things
         )
-        .map(|p|p.clone())
-        .collect();
-
-    Ok(result)
+        .cloned()
+        .collect()
 }
 
-pub fn add_binfolder_to_path_in_shell_scripts(bin_path: &PathBuf) -> Result<()> {
-    let paths = find_shell_scripts_to_be_modified(true)?;
+pub fn add_binfolder_to_path_in_shell_scripts(bin_path: &Path) {
+    let paths = find_shell_scripts_to_be_modified(true);
 
     paths.into_iter().for_each(|p| {
-        add_path_to_specific_file(bin_path, p).unwrap();
+        add_path_to_specific_file(bin_path, &p).unwrap();
     });
-
-    Ok(())
 }
 
-pub fn remove_binfolder_from_path_in_shell_scripts() -> Result<()> {
-    let paths = find_shell_scripts_to_be_modified(false)?;
+pub fn remove_binfolder_from_path_in_shell_scripts() {
+    let paths = find_shell_scripts_to_be_modified(false);
```

Comment:
These two functions cannot error either, so just return the unit type.

Reply:
Same thing, I believe the proper fix is to replace the `unwrap` calls in the function body with `?` and then leave the return type as is.